cosmo_tester/framework/test_hosts.py

```diff
@@ -89,6 +90,66 @@ def assign(
         self.deployment_id = deployment_id
         self.server_id = server_id
 
+    def prepare_for_windows(self, image, user):
+        """Prepare this VM to be created as a windows VM."""
+        add_firewall_cmd = "&netsh advfirewall firewall add rule"
+        password = 'AbCdEfG123456!'
+
+        self.enable_ssh_wait = False
+        self.upload_files = False
+        self.restservice_expected = False
+        self.should_finalize = False
+        self.image_name = image
+        self.username = user
+
+        self.userdata = """#ps1_sysnative
+$PSDefaultParameterValues['*:Encoding'] = 'utf8'
+
+Write-Host "## Configuring WinRM and firewall rules.."
+winrm quickconfig -q
+winrm set winrm/config              '@{{MaxTimeoutms="1800000"}}'
+winrm set winrm/config/winrs        '@{{MaxMemoryPerShellMB="300"}}'
+winrm set winrm/config/service      '@{{AllowUnencrypted="true"}}'
+winrm set winrm/config/service/auth '@{{Basic="true"}}'
+{fw_cmd} name="WinRM 5985" protocol=TCP dir=in localport=5985 action=allow
+{fw_cmd} name="WinRM 5986" protocol=TCP dir=in localport=5986 action=allow
+
+Write-Host "## Setting password for Admin user.."
+$user = [ADSI]"WinNT://localhost/{user}"
+$user.SetPassword("{password}")
+$user.SetInfo()""".format(fw_cmd=add_firewall_cmd,
+                          user=self.username,
+                          password=password)
+
+        self.password = password
+        return password
+
+    @retrying.retry(stop_max_attempt_number=120, wait_fixed=3000)
+    def wait_for_winrm(self):
+        self._logger.info('Checking Windows VM %s is up...', self.ip_address)
+        self.run_windows_command('Write-Output "Testing winrm."',
+                                 powershell=True)
+        self._logger.info('...Windows VM is up.')
+
+    def run_windows_command(self, command, powershell=False,
```

Comment:
probably better if this was just `run_command`?

Reply:
i'm aware `powershell=` isn't an argument for `run_command` but perhaps it is/should be always powershell